File: examples/C#/clonecli1.cs

//
//  Clone client, Model One in C#
//

//  Author:     Yan Cui
//  Email:      theburningmonk@gmail.com

using System;
using System.Collections.Generic;
using System.Text;
using ZeroMQ;

namespace zguide.clonecli1
{
    internal class Program
    {
        public static void Main(string[] args)
        {
            using (var context = ZmqContext.Create())
            {
                using (var subscriber = context.CreateSocket(SocketType.SUB))
                {
                    subscriber.Connect("tcp://localhost:5556");
                    subscriber.Subscribe(Encoding.Unicode.GetBytes(string.Empty));

                    var interrupted = false;
                    var sequence = 0L;
                    var dict = new Dictionary<string, KvMsg>();

                    Console.CancelKeyPress += (s, e) => { interrupted = true; };

                    while (!interrupted)
                    {
                        KvMsg kvmsg;

                        try
                        {
                            kvmsg = KvMsg.Receive(subscriber);
                            Console.WriteLine("Received {0}", kvmsg);
                        }
                        catch (System.Exception)
                        {
                            break;
                        }

                        kvmsg.Store(dict);
                        sequence++;
                    }

                    Console.WriteLine(" Interrupted\n{0} messages in\n", sequence);
                    Console.ReadKey();
                }
            }
        }
    }
}

File: examples/Java/Clonecli1.java
import java.util.HashMap;
import java.util.Map;
import java.util.concurrent.atomic.AtomicLong;

import org.zeromq.ZContext;
import org.zeromq.ZMQ;
import org.zeromq.ZMQ.Socket;

/**
 * Clone client model 1
 * @author Danish Shrestha <dshrestha06@gmail.com>
 *
 */
public class Clonecli1 {
	private static Map<String, Kvsimple> kvMap = new HashMap<String, Kvsimple>();
	private static AtomicLong sequence = new AtomicLong();

	public void run() {
		ZContext ctx = new ZContext();
		Socket subscriber = ctx.createSocket(ZMQ.SUB);
		subscriber.connect("tcp://localhost:5556");
		subscriber.subscribe("".getBytes());

		while (true) {
			Kvsimple kvMsg = Kvsimple.recv(subscriber);
            if (kvMsg == null)
                break;

            Clonecli1.kvMap.put(kvMsg.getKey(), kvMsg);
            System.out.println("receiving " + kvMsg);
            sequence.incrementAndGet();
		}
        ctx.destroy();
	}

	public static void main(String[] args) {
		new Clonecli1().run();
	}
}
